src/audio/component.c

```diff
@@ -227,7 +227,15 @@ void comp_set_period_bytes(struct comp_dev *dev, uint32_t frames,
 
 void sys_comp_init(void)
 {
+	extern unsigned int _comp_init_start, _comp_init_end;
+
 	cd = rzalloc(RZONE_SYS, SOF_MEM_CAPS_RAM, sizeof(*cd));
 	list_init(&cd->list);
 	spinlock_init(&cd->lock);
+
+	void(*comp_init)(void) = (void(*)(void))(&_comp_init_start + 0x2);
+
+	for (; comp_init < (void(*)(void))&_comp_init_end; comp_init += 0x18) {
```

Comment:
looks comp_register() itself is not with so small footprint, 0x18 is xtensa specific? if so, we shouldn't put it in generic src/audio/component.c.

Reply:
Probably not xtensa specific, this just depends on the comp_init functions size. 
comp_register is not a part of this special section so it doesnt matter.

Anyway this magic numbers will be deleted to make it looks much better